presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/operator/aggregation/GeometryStateFactory.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.geospatial.operator.aggregation;
+
+import com.esri.core.geometry.ogc.OGCGeometry;
+import com.facebook.presto.array.ObjectBigArray;
+import com.facebook.presto.geospatial.GeometryType;
+import com.facebook.presto.operator.aggregation.state.AbstractGroupedAccumulatorState;
```

Comment:
@tdcmeehan `AbstractGroupedAccumulatorState` comes from `presto-main`, but plugins cannot depend on that code. `AbstractGroupedAccumulatorState` doesn't have much code/logic in it. Let's just copy-paste it into `GroupedGeometryState`.

```
    public static class GroupedGeometryState
            implements GeometryState, GroupedAccumulatorState
    {
        private final ObjectBigArray<OGCGeometry> array = new ObjectBigArray<>();
        private long size;
        private long groupId;

        @Override
        public final void setGroupId(long groupId)
        {
            this.groupId = groupId;
        }

        @Override
        public OGCGeometry getGeometry()
        {
            return array.get(groupId);
        }
```